ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/LinearRaoProblem.java

```diff
@@ -7,248 +7,147 @@
-            .findFirst()
-            .orElse(null);
+    public MPVariable getRangeActionSetPointVariable(RangeAction rangeAction) {
+        return solver.lookupVariableOrNull(rangeActionSetPointVariableId(rangeAction));
     }
 
-    public List<MPVariable> getNegativeRangeActionVariables() {
-        return negativeRangeActionVariables;
+    public String aboluteRangeActionVariationVariableId(RangeAction rangeAction) {
+        return rangeAction.getId() + SEPARATOR + ABSOLUTE_VARIATION + SEPARATOR + VARIABLE_SUFFIX;
     }
 
-    public MPVariable getNegativeRangeActionVariable(String rangeActionId) {
-        return negativeRangeActionVariables.stream()
-            .filter(variable -> variable.name().equals(getNegativeRangeActionVariableId(rangeActionId)))
-            .findFirst()
-            .orElse(null);
+    public MPVariable addAboluteRangeActionVariationVariable(double lb, double ub, RangeAction rangeAction) {
+        return solver.makeNumVar(lb, ub, aboluteRangeActionVariationVariableId(rangeAction));
     }
 
-    public List<MPVariable> getPositiveRangeActionVariables() {
-        return positiveRangeActionVariables;
+    public MPVariable getAboluteRangeActionVariationVariable(RangeAction rangeAction) {
+        return solver.lookupVariableOrNull(aboluteRangeActionVariationVariableId(rangeAction));
     }
 
-    public MPVariable getPositiveRangeActionVariable(String rangeActionId) {
-        return positiveRangeActionVariables.stream()
-            .filter(variable -> variable.name().equals(getPositiveRangeActionVariableId(rangeActionId)))
-            .findFirst()
-            .orElse(null);
+    private String absoluteRangeActionVariationConstraintId(RangeAction rangeAction, AbsExtension positiveOrNegative) {
+        return rangeAction.getId() + SEPARATOR + ABSOLUTE_VARIATION + positiveOrNegative.toString().toLowerCase() + SEPARATOR + CONSTRAINT_SUFFIX;
     }
 
-    /**
-     * @param cnecId id of the considered cnec
-     * @param minMax can take two values: "min" or "max" depending on the expected MPConstraint
-     * @return the MPConstraint ensuring that the minimal margin is smaller than the
-     * minimum or maximum margin of the considered cnec, or null if this MPConstraint doesn't exist
-     */
-    public MPConstraint getMinimumMarginConstraint(String cnecId, String minMax) {
-        return minimumMarginConstraints.stream()
-            .filter(constraint -> constraint.name().equals(getMinimumMarginConstraintId(cnecId, minMax)))
-            .findFirst()
-            .orElse(null);
+    public MPConstraint addAboluteRangeActionVariationConstraint(double lb, double ub, RangeAction rangeAction, AbsExtension positiveOrNegative) {
+        return solver.makeConstraint(lb, ub, absoluteRangeActionVariationConstraintId(rangeAction, positiveOrNegative));
+    }
+
+    public MPConstraint getAboluteRangeActionVariationConstraint(RangeAction rangeAction, AbsExtension positiveOrNegative) {
```

Comment:
You can rename all abolute methods in absolute

Reply:
Abolutely !